src/main/scala/com/normation/plugins/datasources/api/DataSourceApi9.scala

```diff
@@ -107,17 +110,36 @@ class DataSourceApi9 (
 
       apiService.reloadDataOneNodeFor(actor, NodeId(nodeId), DataSourceId(datasourceId))
 
-      toJsonResponse(None, JString(s"Data for node '${nodeId}', for all configured data sources, is going to be updated"))
+      toJsonResponse(None, JString(s"Data for node '${nodeId}', for data source '${datasourceId}', is going to be updated"))
+    }
+
+    case Post( "clear" :: datasourceId :: Nil, req) => {
+      implicit val prettify = extractor.extractPrettify(req.params)
+      implicit val action = "clearValueOneDatasourceAllNodes"
+      val actor = RestUtils.getActor(req)
+
+      apiService.clearDataAllNodesFor(actor, DataSourceId(datasourceId)) match {
+        case Full(_)     => toJsonResponse(None, JString(s"Data for all nodes, for data source '${datasourceId}', cleared"))
+        case eb:EmptyBox => toJsonError(None, JString((eb ?~! "Could not clear data source property").messageChain))
+      }
     }
 
+    case Post("clear" :: datasourceId :: "node" :: nodeId :: Nil, req) => {
+      implicit val prettify = extractor.extractPrettify(req.params)
+      implicit val action = "clearValueOneDatasourceOneNode"
+      val actor = RestUtils.getActor(req)
+
+      apiService.clearDataOneNodeFor(actor, NodeId(nodeId), DataSourceId(datasourceId)) match {
+        case Full(_)     => toJsonResponse(None, JString(s"Data for node '${nodeId}', for data source '${datasourceId}', cleared"))
+        case eb:EmptyBox => toJsonError(None, JString((eb ?~! "Could not clear data source property").messageChain))
+      }
+    }
 
     case Post( "reload" :: Nil, req) => {
       implicit val prettify = extractor.extractPrettify(req.params)
       implicit val action = "reloadAllDatasourcesAllNodes"
       val actor = RestUtils.getActor(req)
 
-      apiService.reloadDataAllNodes(actor)
```

Comment:
WHy do you remove it ?

Reply:
hum. No idea, it was not supposed to be done.